app/policies/mhv_medical_records_policy.rb

```diff
@@ -1,9 +1,38 @@
 # frozen_string_literal: true
 
+require 'medical_records/user_eligibility/client'
+
 MHVMedicalRecordsPolicy = Struct.new(:user, :mhv_medical_records) do
   MR_ACCOUNT_TYPES = %w[Premium].freeze
 
   def access?
-    MR_ACCOUNT_TYPES.include?(user.mhv_account_type) && user.va_patient?
+    if Flipper.enabled?(:mhv_medical_records_new_eligibility_check)
+      client = UserEligibility::Client.new(session: { user_id: user.mhv_correlation_id, icn: user.icn })
```

Comment:
Seems like there should be a rescue clause here incase of issue in the client requests performed here

Reply:
Hi Lindsey, thank you for your review. I moved the error handling block to the access method so it would catch client failures as well, and added the error message to the logged details. Let me know if you see anything else that could be improved. Thanks!